### PR TITLE
[DEV-2994] Enable Feast integration by default

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR enable Feast integration by default.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/service/offline_store_feature_table_manager.py [hunk 1]
```diff
@@ -419,10 +429,33 @@ async def _get_required_aggregate_result_tables(
...
+        fl_has_been_enabled = False
         async for feature_list_dict in self.feature_list_service.iterate_online_enabled_feature_lists_as_dict():
-            feature_lists.append(FeatureListModel(**feature_list_dict))
+            feature_list = FeatureListModel(**feature_list_dict)
+            if (
+                feature_list_to_online_disable
+                and feature_list.id == feature_list_to_online_disable.id
+            ):
+                # skip the feature list to be disabled
+                continue
```
Comment (YS-L) [inline featurebyte/service/offline_store_feature_table_manager.py:446]: We need this check because now the feature list's `enabled` flag is updated at the very last step of the deployment update process, is that correct?

File: tests/integration/query_graph/test_online_serving.py [hunk 2]
```diff
@@ -89,6 +90,9 @@ def features_fixture(event_table, source_type):
     return features
 
 
+@pytest.mark.parametrize(
+    "source_type", SNOWFLAKE_SPARK_DATABRICKS, indirect=True  # skip unity catalog: DEV-3109
```
Comment (YS-L) [inline tests/integration/query_graph/test_online_serving.py:94]: Is this an issue that rarely occurs? I think we unconditionally skip the `test / test-integration-databricks (pull_request) Skipped` suite now, so `SNOWFLAKE_SPARK_DATABRICKS` is effectively the same as `SNOWFLAKE_SPARK`.

File: featurebyte/models/deployment.py [hunk 3]
```diff
@@ -58,4 +58,4 @@ class FeastIntegrationSettings(BaseSettings):
     Feast integration settings
     """
 
-    FEATUREBYTE_FEAST_INTEGRATION_ENABLED: bool = False
+    FEATUREBYTE_FEAST_INTEGRATION_ENABLED: bool = True
```
Comment (YS-L) [inline featurebyte/models/deployment.py:61]: All the integration tests including databricks_unity run with this flag enabled, is that correct?

File: featurebyte/service/offline_store_feature_table_manager.py (hunk 1 above)
Comment (gee-senbong) [reply featurebyte/service/offline_store_feature_table_manager.py:446]: Yes, the feature list's attribute is updated at later step in `DeployService`.
```

        if feast_registry:
            await self.feast_integration_service.handle_online_enabled_features(  # this calls offline store ft manager 
                features=features_offline_feature_table_to_update,
                feature_list_to_online_enable=feature_list,
                update_progress=get_ranged_progress_callback(
                    self.task_progress_updater.update_progress, 72, 98
                ),
            )
            await self._update_progress(
                99, f"Updating deployed feature list ({feature_list.name}) ..."
            )
            await self.feast_integration_service.handle_deployed_feature_list(   # feature list attr updated here
                feature_list=feature_list, online_enabled_features=features
            )
```

File: tests/integration/query_graph/test_online_serving.py (hunk 2 above)
Comment (gee-senbong) [reply tests/integration/query_graph/test_online_serving.py:94]: I think this issue occurs deterministically. After we fixed other flaky tests.

File: featurebyte/service/offline_store_feature_table_manager.py (hunk 1 above)
Comment (YS-L) [reply featurebyte/service/offline_store_feature_table_manager.py:446]: I see, in that case it seems like an existing bug if I understood correctly. Good to fix it 👍

File: featurebyte/models/deployment.py (hunk 3 above)
Comment (gee-senbong) [reply featurebyte/models/deployment.py:61]: Yes, all the tests will assume Feast integration is enabled by default unless we explicitly mock it to false.

File: tests/integration/query_graph/test_online_serving.py (hunk 2 above)
Comment (YS-L) [reply tests/integration/query_graph/test_online_serving.py:94]: Oh I meant whether it's an issue that will be easily encountered by users, or more a corner case that rarely occurs in real life.

File: featurebyte/models/deployment.py (hunk 3 above)
Comment (YS-L) [reply featurebyte/models/deployment.py:61]: Got it, wanted to confirm since based on our offline discussion the databricks unity integration tests would run with the flag disabled. Seems like that is no longer required.

**Review by YS-L — APPROVED** [review]

File: featurebyte/service/offline_store_feature_table_manager.py (hunk 1 above)
Comment (gee-senbong) [reply featurebyte/service/offline_store_feature_table_manager.py:446]: Previously we don't rely on the feature list attribute to check whether the Feast integration is enabled or not. Strictly speaking, it is not a bug.

File: tests/integration/query_graph/test_online_serving.py (hunk 2 above)
Comment (gee-senbong) [reply tests/integration/query_graph/test_online_serving.py:94]: I suspect we need to fix it. But it seems like `Feast` doesn't expect `np.float32` type (based on my rough understanding of the exception message show in the ticket). https://featurebyte.atlassian.net/browse/DEV-3109

File: featurebyte/models/deployment.py (hunk 3 above)
Comment (gee-senbong) [reply featurebyte/models/deployment.py:61]: Ya, it is no longer required as this is the only integration test that failed with Feast integration with unity catalog at the moment.
